larreco/QuadVtx/HeatMap.h

```diff
@@ -0,0 +1,35 @@
+#ifndef LARRECO_QUADVTX_HEATMAP_H
+#define LARRECO_QUADVTX_HEATMAP_H
+
+#include <vector>
+#include <memory>
+
+class TH2F;
+
+namespace quad
+{
+  class HeatMap
+  {
+  public:
+    HeatMap(int _Nz, float _minz, float _maxz,
+            int _Nx, float _minx, float _maxx);
+
+    std::unique_ptr<TH2F> AsTH2() const;
+
+    inline float ZBinCenter(int iz) const {return minz + (iz+.5)*(maxz-minz)/Nz;}
+    inline float XBinCenter(int ix) const {return minx + (ix+.5)*(maxx-minx)/Nx;}
+    inline int ZToBin(float z) const {return fast_floor((z-minz)/(maxz-minz)*Nz);}
+    inline int XToBin(float x) const {return fast_floor((x-minx)/(maxx-minx)*Nx);}
+
+    float minz, minx, maxz, maxx;
+    int Nx, Nz;
+
+    std::vector<float> map;
```

Comment:
Can these data members be made private?  Not generally a good idea to expose member data of classes.

Reply:
I'm against adding trivial accessors on basic struct-like objects like this. Turns out I can make them const, which should address the main concern.

Reply:
Ah...yes, this is the frustration about classes vs. structs.  If the type is intended to be struct-like, then keeping them public is the way to go.  That would argue toward changing from `class` to `struct`, as the code is more idiomatic of structs.  But it's your call if you want to do that.